### DISCUSSION TITLE
Should Fn.Ref take a Text instead of a Fn?

### DISCUSSION THREAD

**Discussion by endgame** [jcouyang/dhall-aws-cloudformation#8] [General]
I find my self writing `Fn.Ref (Fn.String ...)` an awful lot, and I don't think I've ever seen an instance where `Fn::Ref` was taking a non-literal string as an argument. Does CloudFormation support recursive `Fn::Ref`s, and if not, should this be tighened to `Ref : Text -> Fn`?

Alternatively, would you take a PR that added `Fn.refText : Text -> Fn/Type` for the common case, in similar spirit to `Fn.renderText`?

**Comment by jcouyang** [comment]
Yeah, you're right, based on the doc it should only take string, should be safe to change Ref type to `Text -> Fn`

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-ref.html
> You can't use any functions in the Ref function. You must specify a string that's a resource logical ID.

**Comment by endgame** [comment]
Saw the new release, thank you. I also saw that `README.md` still has an example that uses `String` when constructing a ref - I think this needs to be fixed too.
